google-cloud-firestore/src/test/java/com/google/cloud/firestore/LocalFirestoreHelper.java

```diff
@@ -970,18 +978,88 @@ public static String autoId() {
   }
 
   private static Map<String, Object> fromJsonString(String json) {
-    try {
-      ObjectMapper mapper = new ObjectMapper();
-      return mapper.readValue(json, new TypeReference<Map<String, Object>>() {});
-    } catch (IOException e) {
-      throw new RuntimeException(e);
-    }
+    Type type = new TypeToken<Map<String, Object>>() {}.getType();
+    GsonBuilder gsonBuilder = new GsonBuilder();
+    gsonBuilder.registerTypeAdapterFactory(CustomObjectTypeAdapter.FACTORY);
+    Gson gson = gsonBuilder.create();
+    return gson.fromJson(json, type);
   }
 
   public static Map<String, Object> fromSingleQuotedString(String json) {
     return fromJsonString(json.replace("'", "\""));
   }
 
+  static class CustomObjectTypeAdapter extends TypeAdapter<Object> {
+    public static final TypeAdapterFactory FACTORY =
+        new TypeAdapterFactory() {
+          @SuppressWarnings("unchecked")
+          @Override
+          public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> type) {
+            if (Map.class.isAssignableFrom(type.getRawType())) {
+              return (TypeAdapter<T>) new CustomObjectTypeAdapter();
+            }
+            return null;
+          }
+        };
+
+    private final TypeAdapter<Object> delegate = new Gson().getAdapter(Object.class);
+
+    @Override
+    public Object read(JsonReader in) throws IOException {
+      JsonToken token = in.peek();
+      switch (token) {
+        case BEGIN_ARRAY:
+          List<Object> list = new ArrayList<Object>();
+          in.beginArray();
+          while (in.hasNext()) {
+            list.add(read(in));
+          }
+          in.endArray();
+          return list;
+
+        case BEGIN_OBJECT:
+          Map<String, Object> map = new LinkedTreeMap<String, Object>();
+          in.beginObject();
+          while (in.hasNext()) {
+            map.put(in.nextName(), read(in));
+          }
+          in.endObject();
+          return map;
+
+        case STRING:
+          return in.nextString();
+
+        case NUMBER:
```

Comment:
Can you simply use doubles here?

Reply:
Sure, I will remove custom object adapter altogether and will update the tests accordingly because existing tests need numbers to be Integers, long and doubles